### ISSUE TITLE
No longer works with "api" GitHub Enterprise subdomain

### ISSUE THREAD

**Issue by oschwald** [google/go-github#1552]
As of #1165, `go-github` no longer works if the `api` GitHub Enterprise subdomain is used, which broke our use of it. The `/api/v3` suffix should only be added if the domain does not start with `api.`

```
$ curl -u <user>:<token> https://api.github.maxmind.com/api/v3/repos/maxmind/mm_website
{
  "message": "Not Found",
  "documentation_url": "https://developer.github.com/enterprise/2.20/v3"
}

$ curl -u <user>:<token> https://api.github.maxmind.com/repos/maxmind/mm_website
{
  <expected data>
}

$ curl -u <user>:<token> https://github.maxmind.com/api/v3/repos/maxmind/mm_website
{
  <expected data>
}
```

**Comment by gmlewis** [comment]
Hmmm... I'm not sure I understand your example...  Why are you using `https://api.github.maxmind.com` when the third example is actually working properly (which this repo supports) - `https://github.maxmind.com` ?

**Comment by oschwald** [comment]
The `api` subdomain is a feature of GitHub Enterprise and worked fine with `go-github` through version v28. In #1165, `go-github` started appending `/api/v3` to the end of Enterprise base URLs that didn't already have the suffix. This broke any existing users that were using the `api` subdomain.

We noticed this when upgrading from v28 to v32. The cause of the breakage was not immediately obvious. It was not listed as one of the breaking changes in the release notes and, from the PR, I don't think the intent was to break this particular use.

We did end up removing the `api` subdomain to work around this issue. However, the change in #1165 could be tweaked slightly to only add the suffix to base URLs that don't already have the suffix and are not using the `api` subdomain. This would prevent other users from running into this.

**Comment by gmlewis** [comment]
OK, I apologize for the breakage, as I was unaware of the `api` subdomain.
Yes, please feel free to write a PR to fix this for other users, and thank you for alerting us to this breakage.

Or if you would like other volunteers to address this issue, that is fine too, and I can open it up for others to contribute.  It's your call.

**Comment by chughpiyush** [comment]
I can work on this one.

**Comment by chughpiyush** [comment]
Hi @gmlewis ,
Is this issue still something we would like to fix? If yes, I can have a PR ready quickly.

**Comment by gmlewis** [comment]
Yes, please, @chughpiyush. Please make sure to fully unit test the various scenarios described above.
